lib/rtcpeerconnection/chrome.js

```diff
@@ -112,22 +117,69 @@ function ChromeRTCPeerConnection(configuration) {
     // the ontrack property of the RTCPeerConnection.
   };
 
+  peerConnection.addStream(this._localStream);
   util.proxyProperties(PeerConnection.prototype, this, peerConnection);
 }
 
 inherits(ChromeRTCPeerConnection, EventTarget);
 
-ChromeRTCPeerConnection.prototype.addStream = function addStream(stream) {
-  if (this._localStreams.has(stream)) {
+// NOTE(mmalavalli): This shim supports our limited case of adding
+// all MediaStreamTracks to one MediaStream. It has been implemented this
+// keeping in mind that this is to be maintained only until "addTrack" is
+// supported natively in Chrome.
+ChromeRTCPeerConnection.prototype.addTrack = function addTrack() {
+  var args = [].slice.call(arguments);
+  if ('addTrack' in PeerConnection.prototype) {
+    return this._peerConnection.addTrack.apply(this._peerConnection, args);
+  }
+  var track = args[0];
+
+  if (this._peerConnection.signalingState === 'closed') {
+    throw new Error('Cannot add MediaStreamTrack [' + track.id + ', '
+      + track.kind + ']: RTCPeerConnection is closed');
+  }
+  if (this._senders.has(track.id)) {
+    throw new Error('Cannot add MediaStreamTrack [' + track.id + ', '
+      + track.kind + ']: RTCPeerConnection already has it');
+  }
+  this._peerConnection.removeStream(this._localStream);
+  this._localStream.addTrack(track);
+  this._peerConnection.addStream(this._localStream);
+
+  var sender = new RTCRtpSenderForLegacyWebRTC(track);
+  this._senders.set(track.id, sender);
+  return sender;
+};
+
+// NOTE(mmalavalli): This shim supports our limited case of removing
+// MediaStreamTracks from one MediaStream. It has been implemented this
+// keeping in mind that this is to be maintained only until "removeTrack" is
+// supported natively in Chrome.
+ChromeRTCPeerConnection.prototype.removeTrack = function removeTrack(sender) {
+  if ('removeTrack' in PeerConnection.prototype) {
+    this._peerConnection.removeTrack(sender);
     return;
   }
-  this._localStreams.add(stream);
-  this._peerConnection.addStream(stream);
+
+  var track = sender.track;
+  if (this._peerConnection.signalingState === 'closed') {
+    throw new Error('Cannot remove MediaStreamTrack [' + track.id +
+      ', ' + track.kind + ']: RTCPeerConnection is closed');
+  }
+  if (!this._senders.has(track.id)) {
+    return;
+  }
+  this._senders.delete(track.id);
```

Comment:
Can we do

```js
if (this._senders.delete(track.id)) {
  this._peerConnection.removeStream(this._localStream);
  // ...
}
```

Instead of the separate `has` check?